web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -482,16 +484,24 @@ else if (SIMPLE_SETUP.equals(page)) {
 				renderTemplate(INSTALL_METHOD, referenceMap, httpResponse);
 				return;
 			}
-			wizardModel.databaseConnection = httpRequest.getParameter("database_connection");
-
-			wizardModel.createDatabaseUsername = Context.getRuntimeProperties().getProperty("connection.username",
-				wizardModel.createDatabaseUsername);
-			
-			wizardModel.createUserUsername = wizardModel.createDatabaseUsername;
 			
+			String databaseType = httpRequest.getParameter("database_type");
+			if (databaseType != null) {
+				wizardModel.databaseType = databaseType;
+				if (DATABASE_MYSQL.equals(databaseType)) {
+					wizardModel.databaseConnection = DEFAULT_MYSQL_CONNECTION;
+					wizardModel.createDatabaseUsername = Context.getRuntimeProperties().getProperty("connection.username", wizardModel.createDatabaseUsername);
+					checkForEmptyValue(wizardModel.databaseRootPassword, errors, ErrorMessageConstants.ERROR_DB_PSDW_REQ);
+				} else if (DATABASE_POSTGRESQL.equals(databaseType)) {
+					wizardModel.databaseConnection = DEFAULT_POSTGRESQL_CONNECTION;
+					String postgresUsername = httpRequest.getParameter("create_database_username");
+					wizardModel.createDatabaseUsername = StringUtils.hasText(postgresUsername) ? 
```

Comment:
Is there a reason why you also do not do this for mysql?

Reply:
Fair enough!

Reply:
noted!